platforms/access-bridge/src/services/stripe-service.ts

```diff
@@ -0,0 +1,60 @@
+import Stripe from 'stripe';
+
+import { STRIPE_SECRET } from '../app-config.js';
+
+export type StripeProduct = Stripe.Product & {
+  prices: Stripe.Price[];
+};
+
+/**
+ * Service class responsible for interacting with the Stripe API to fetch products.
+ */
+export class StripeService {
+  private stripe: Stripe;
+
+  constructor() {
+    this.stripe = new Stripe(STRIPE_SECRET, {
+      apiVersion: '2024-06-20',
+    });
+  }
+
+  /**
+   * Retrieves Stripe products with prices and filters them based on the provided productIds.
+   * @param productIds The array of product IDs to filter the products.
+   * @returns A Promise resolving to an array of filtered ProductsWithMetadata objects.
+   */
+  async getProductsWithPrices(productIds: string[]): Promise<StripeProduct[]> {
+    if (!productIds.length) {
+      return [];
+    }
+
+    // Fetch all products (pagination might be necessary for large datasets)
+    const allProducts = await this.stripe.products.list();
+    const filteredProducts = allProducts.data.filter((product) => productIds.includes(product.id));
+
+    if (filteredProducts.length === 0) {
+      return [];
+    }
+
+    // Retrieve prices for each filtered product
+    const productsWithPrices = await Promise.all(
+      filteredProducts.map(async (product) => {
```

Comment:
Any thoughts on sorting? Do we just let the FE handle that?

Reply:
By default they are sorted by date of creation. If we need additional logic it can always be managed on the FE.